core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -585,7 +621,12 @@ private void initializeWriter(StoreConfiguration cfg, CacheWriter writer, Initia
       if (writer != null) {
          if (writer instanceof DelegatingCacheWriter)
```

Comment:
@wburns I never quite figured out what what the use case for singleton stores is, but reading the javadoc of `SingletonCacheWriter` again I'm pretty sure it only makes sense with a shared store. There's no `SingletonCacheLoader` to delegate reads to the coordinator, after all...